Subject: Fix crash when path animation data is empty
From: Subhransu Mohanty <sub.mohanty@samsung.com>
Date: Mon, 25 Jan 2021 10:47:25 +0900
Bug: https://github.com/Samsung/rlottie/issues/460
Bug-Debian: https://bugs.debian.org/974095
Bug-Debian: https://bugs.debian.org/994614
Origin: https://github.com/Samsung/rlottie/commit/1cb2021d6883ebe41c17e710fc90a225f038cb51
Last-Update: Sat, 27 Nov 2021 16:41:51 +0300

---
 src/lottie/lottiemodel.h | 5 +++++
 1 file changed, 5 insertions(+)

--- a/src/lottie/lottiemodel.h
+++ b/src/lottie/lottiemodel.h
@@ -124,6 +124,11 @@ public:
     static void lerp(const LottieShapeData& start, const LottieShapeData& end, float t, VPath& result)
     {
         result.reset();
+        // test for empty animation data.
+        if (start.mPoints.empty() || end.mPoints.empty())
+        {
+            return;
+        }
         auto size = std::min(start.mPoints.size(), end.mPoints.size());
         /* reserve exact memory requirement at once
          * ptSize = size + 1(size + close)
